src/libraries/System.Private.CoreLib/src/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -281,7 +281,7 @@ public bool TryPush(T[] array)
                 // Try to push on to the associated stack first.  If that fails,
                 // round-robin through the other stacks.
                 LockedStack[] stacks = _perCoreStacks;
-                int index = Thread.GetCurrentProcessorId() % stacks.Length;
+                int index = Thread.GetCurrentProcessorId() % s_lockedStackCount; // mod by constant in tier 1
```

Comment:
if both are never negative you can add an `uint` cast - it's optimized even better, e.g. [sharplab.io](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYRoG8aHuGBLAOwwMAsuQAUAoQgCUXHp2o8lTAOwMEDAKQNyABgDcc7gF8lRhuckjSEwetmL55pQHoXDAJIMA1vwgB3BgwAC2wpPlwGfhgANxgoKJgAczDeOIZg+JgQZx5iNVsMaTExAFdJaQ1tPWlDRxMaYyA=)

Reply:
Cool, thanks